src/sys/sockets.nim

```diff
@@ -883,3 +884,82 @@ proc accept*(l: AsyncListener[TCP]): tuple[conn: AsyncConn[TCP], remote: IPEndpo
 proc localEndpoint*(l: AsyncListener[TCP] | Listener[TCP]): IPEndpoint {.raises: [OSError].} =
   ## Obtain the local endpoint of `l`.
   tcpLocalEndpoint()
+
+proc connectUnix*(path: string): Conn[Unix]
+                 {.raises: [OSError].} =
+  ## Create a connection to the Unix socket at `path`.
+  ##
+  ##
+  ## Unix sockets are currently only supported on POSIX-like
+  ## platforms (this does not include Windows).
+  ## Unix socket implementations can differ in the maximum
+  ## supported length of path elements as well as the total
+  ## path length.
+  ## On Linux the maximum path length is hardcoded to 107
+  ## characters.
+  unixConnect()
```

Comment:
While I'm not a big fan of doing this, I think the solution would be to put all Unix sockets functions in a `when defined(posix)` until we gain a Windows implementation